app/controllers/contact-details.ts

```diff
@@ -0,0 +1,95 @@
+import { NextFunction, Request, Response, Router } from 'express';
+import { paths } from '../paths';
+import { emailValidation, phoneValidation } from '../utils/fields-validations';
+
+function hasSelections(data: []) {
+  return data !== undefined;
+}
+
+function validate(request: any) {
+
+  let validationErrors = {};
+  const selections = request.selections;
+
+  if (selections.includes('email')) {
+    // Validate email
+    const validationResult = emailValidation(request);
+    if (validationResult != null) {
+      validationErrors = { ...validationErrors, ...validationResult };
+    }
+  }
+
+  if (selections.includes('text-message')) {
+    // Validate phone number
+    const validationResult = phoneValidation(request);
+    if (validationResult != null) {
+      validationErrors = { ...validationErrors, ...validationResult };
+    }
+  }
+  return Object.keys(validationErrors).length === 0 ? null : validationErrors;
+}
+
+function getContactDetails(req: Request, res: Response, next: NextFunction) {
+  try {
+    // TODO: get email and phone from idam
+    const data = {
+      email: { value: '' },
+      textMessage: { value: '' }
+    };
+    res.render('contact-details.njk', { data: data });
+  } catch (error) {
+    next(error);
+  }
+}
+
+function postContactDetails(req: Request, res: Response, next: NextFunction) {
+  const request = req.body;
+
+  const flattenToErrorList = (validationErrors: {}) => {
+    const errorList = [];
+    for (let key in validationErrors) {
+      errorList.push(validationErrors[key]);
+    }
+    return errorList;
+  };
+
+  try {
+    const data = {
+      email: { value: request['email-value'] },
+      textMessage: { value: request['text-message-value'] }
+    };
+
+    if (!hasSelections(request.selections)) {
+      return res.render('contact-details.njk', { data: data, errors: { selections: 0 } });
+    }
+    const validationErrors = validate(request);
+    if (validationErrors != null) {
+      return res.render('contact-details.njk', {
+        data: data,
+        errors: validationErrors,
+        errorList: flattenToErrorList(validationErrors)
```

Comment:
`Object.values(validationErrors)` does the trick as well...

Reply:
Thanks ! 👍